staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -158,3 +159,82 @@ func ValidateCustomResourceDefinitionNames(names *apiextensions.CustomResourceDe
 
 	return allErrs
 }
+
+// ValidateCustomResourceDefinitionSchema statically validates
+func ValidateCustomResourceDefinitionSchema(JSONSchema *apiextensions.JSONSchemaProps, fldPath *field.Path) field.ErrorList {
```

Comment:
Can we try to convert the ref to an openapi ref here as well for validation? Then we can be sure that the actual conversion later won't error out.